### PR TITLE
Standardised configurator version naming.

### PR DESCRIPTION
**Pull request by mikeller**
Changes:
- [x] standardise the version string used for the configurator;
- [x] for debug builds, add a suffix of `-debug-<revision hash>` to make them easier to identify;
- [x] split up the version line in the log to make it reatable on mobile

image

This is in preparation of a release of configurator to the Google Play Store.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by haslinghuis — DISMISSED** [review]

File: gulpfile.js
Comment (haslinghuis) [inline gulpfile.js:97]: The parameter (true) is not used in cordova_dist(true)

File: gulpfile.js
Comment (haslinghuis) [inline gulpfile.js:99]: The parameter (true) is not used in cordova_dist(true)

File: gulpfile.js
Comment (haslinghuis) [inline gulpfile.js:89]: Should this be isDebugBuild?

File: gulpfile.js
Comment (haslinghuis) [inline gulpfile.js:93]: Should this be isDebugBuild?

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: gulpfile.js
Comment (McGiverGim) [inline gulpfile.js:31]: We usually use this require as an "import" of the library, and we "instantiate" it later when needed. You are "instantiating" it at this point. 
Are we going to go to this way now? I think is good to have all in the same style if possible.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: gulpfile.js
Comment (mikeller) [reply gulpfile.js:27]: It seems to be the default for most packages these days that their singleton instance does not need to be instanciated, as it is returned directly by the call to `require()` - `simple-git` is an exception to this. Not sure if it is a better approach to instanciate the singleton where the package is used, which will likely be inside a function call, and risk being run multiple times if the function is called multiple times.

**Comment by mikeller** [comment]
Update: I have reworked this quite drastically - now all of the version information is injected into the version of package.json that is included in the application. This means that `version.json` is no longer needed and could be eliminated. For the android build, `manifest.json`, containing a copy of `package.json` had to be added, as this is where cordova looks for the information returned by `getManifest()`.

**Review by Asizon — APPROVED** [review]

**Review by ctzsnooze — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> PASS
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by haslinghuis** [comment]
@McGiverGim are you ok with this now?

**Review by McGiverGim — APPROVED** [review]
I can't do a good review now, but I approve it. If something is wrong it can be fixed later.

**Comment by blckmn** [comment]
AUTOMERGE: (PASS)
 - github identifies PR as mergeable -> PASS
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

**Pull request merged**
